### PR TITLE
fix weight setting not working for lines

### PR DESCRIPTION
**Pull request by thor85**
This fixes the line weight bug mentioned in #90.

### CODE REVIEW

File: src/lines.ts [hunk 1]
```diff
@@ -215,7 +216,10 @@ export class Lines extends Base<ILinesSettings> {
       for (let yOffset = -weight; yOffset < weight; yOffset += 0.5) {
         for (let xOffset = -weight; xOffset < weight; xOffset += 0.5) {
           // -- set base matrix to translate canvas pixel coordinates -> webgl coordinates
-          mapMatrix.translateMatrix(-offset.x + (xOffset / scale), -offset.y + (yOffset / scale));
+          mapMatrix
+            .setSize(canvas.width, canvas.height)
```
Comment (robertleeplummerjr) [inline src/lines.ts:220]: I don't believe we need to re-set the size, as it wouldn't have changed between when it is set above and here.

File: src/lines.ts [hunk 2]
```diff
@@ -201,11 +201,12 @@ export class Lines extends Base<ILinesSettings> {
...
     gl.viewport(0, 0, canvas.width, canvas.height);
     gl.vertexAttrib1f(this.aPointSize, pointSize);
-    mapMatrix
-      .setSize(canvas.width, canvas.height)
-      .scaleMatrix(scale);
+
     if (zoom > 18) {
-      mapMatrix.translateMatrix(-offset.x, -offset.y);
+      mapMatrix
+        .setSize(canvas.width, canvas.height)
```
Comment (robertleeplummerjr) [inline src/lines.ts:207]: I believe we still would want to change the size, if we are zoomed to greater than 18 or not.

File: src/lines.ts (hunk 1 above)
Comment (robertleeplummerjr) [reply src/lines.ts:220]: The same for scale, it wouldn't have changed as well.

**Review by robertleeplummerjr — CHANGES_REQUESTED** [review]
I'm not seeing a clear fix here.  Can you explain?

File: src/lines.ts (hunk 1 above)
Comment (thor85) [reply src/lines.ts:220]: No, the size and scale has not changed. But if you do not reset them, then each translateMatrix() in the loop is appended to the previous matrix. These webgl matrices are not my expertice, but have a look at item 12-15 in this output of mapMatrix.

Before:
glify_mapmatrix_before

After my change:
glify_mapmatrix_after

Running the example I can see that this fixes the problem for me, but there may be better ways of fixing it.

File: src/lines.ts (hunk 2 above)
Comment (thor85) [reply src/lines.ts:207]: Yes, we change size for each alternative. I just moved setSize() and scaleMatrix() into each branch to reset the matrix in the loops. For zoom > 18 there should be no change.

**Comment by danwild** [comment]
bump 😬

**Comment by robertleeplummerjr** [comment]
I'm unit testing the whole framework, and am slowly making my way to this area.  I want to reproduce this issue, so we can see the exactly right way fix it.

**Comment by robertleeplummerjr** [comment]
I ended up applying this fix on master, and I thank you for finding it.  I was worried about performance, and I checked it, 1ms was what this takes with a weight of 50, lol.  What was I worried about?

Again ty.

**Comment by robertleeplummerjr** [comment]
For continuity, commit is here: https://github.com/robertleeplummerjr/Leaflet.glify/commit/647b8404706228e5b3a4d71b125b4af75c354498

**Pull request closed without merging**